### DISCUSSION TITLE
Having problems with groups and page permissions

### DISCUSSION THREAD

**Discussion by ghost** [requarks/wiki#5337] [Help / Questions] [answered]
Hello,

I have 4 virtual folders:
folder A
folder B
folder C
folder D

I have 2 groups:
Local
Programmers

I'd like group Local to be able to see only folders A, B, and C, but group Programmers to see all folders. I'm having trouble achieving this.

Keep in mind that a programmer would be in both groups.

I tried creating such page rules for groups:

Local
![image](https://user-images.githubusercontent.com/5992169/170477382-548fc7c8-3ae3-4e37-b8e0-74c33534f606.png)

Programmers
![image](https://user-images.githubusercontent.com/5992169/170477465-831d0ea6-d817-46dd-9c46-ba8f923d04aa.png)

According to documentation, Local group's deny rule will override Programmers group's role.

What I want to achieve:

if a person is in group Local, he will only see folders A, B and C
if a person is in both Local and Programmers groups, he will see folders A, B, C and D

I tried playing with different paths and path matching, but to no avail.

**Comment by NGPixel** [comment, marked as answer]
This is a limitation of the current rules system and is being addressed in v3.

As a workaround, I would suggest removing a character from the **Local** deny rule so that the **Programmers** allow rule is more specific.

Local
- Allow > Path starts with... > `/`
- Deny > Path starts with... > `/programuotojam`

Programmers
- Allow > Path starts with... > `/programuotojams`

**Reply by ghost** [reply]
Thank you.

**Reply by ghost** [reply]
Does user need to relog for page rules to take effect?

**Reply by NGPixel** [reply]
No.

**Reply by ghost** [reply]
Your suggestion works, but user HAS to relog for rules to take effect.

Just refreshing does not work.

**Reply by NGPixel** [reply]
Are you sure you didn't change the groups assigned to the user as well? Page rules are not part of the token, only their group assignments.

**Reply by ghost** [reply]
Ah, so that's what you meant.

Yes, I did change groups, I tried assigning and removing a group to see if your solution works.

And it does :)
